pkg/sqs/sqs.go

```diff
@@ -88,14 +87,7 @@ func (s *AWSSqsSource) Read(_ context.Context, readRequest sourcesdk.ReadRequest
 
 	ctx, cancel := context.WithTimeout(context.Background(), readRequest.TimeOut())
 	defer cancel()
-	// If we have un-acked data (data which is received but yet to be deleted from the queue
-	count, err := s.getApproximateMessageCount(approximateNumberOfMessagesNotVisible)
-	if err != nil {
-		log.Fatalf("error getting approximate number of messages not visible from queue %s", err)
-	}
-	if count > 0 {
-		return
-	}
```

Comment:
On the platform side, the source forwarder doesn't read the next batch until the previous one gets acknowledged. This change breaks the contract. As a result, the source forwarder can read duplicate data and rely on the platform to de-duplicate.

Since we are trying to avoid global NotVisible messages blocking all source pods from reading. Instead of consulting SQS, can we keep a local copy of the un-acked messages on each of the pods? If the list is not empty, we return. That way we both follow the contract and prevent reading duplicate data.